doc/source/release/1.15.0-notes.rst

```diff
@@ -87,11 +315,76 @@ Deprecated features
 ******************************
 Backwards incompatible changes
 ******************************
+- ``interpolate.BSpline.integrate`` output is now always a numpy array.
+  Previously, for 1D splines the output was a python float or a 0D array
+  depending on the value of the ``extrapolate`` argument.
+- `scipy.stats.wilcoxon` now respects the ``method`` argument provided by the
+  user. Previously, even if ``method='exact'`` was specified, the function
+  would resort to ``method='approx'`` in some cases.
 
 *************
 Other changes
 *************
-
+- `scipy.stats.bootstrap` now emits a ``FutureWarning`` if the shapes of the
+  input arrays do not agree. Broadcast the arrays to the same batch shape
+  (i.e. for all dimensions except those specified by the ``axis`` argument)
+  to avoid the warning. Broadcasting will be performed automatically in the
+  future.
+- SciPy endorsed `SPEC-7 <https://scientific-python.org/specs/spec-0007/>`_,
+  which proposes a ``rng`` argument to control pseudorandom number generation
+  (PRNG) in a standard way, replacing legacy arguments like ``seed`` and
+  ``random_sate``. In many cases, use of ``rng`` will change the behavior of
+  the function unless the argument is already an instance of
+  ``numpy.random.Generator``.
+
+  - Effective in SciPy ``1.15.0``:
+
+    - The ``rng`` argument has been added to the following functions:
+      `scipy.cluster.vq.kmeans`, `scipy.cluster.vq.kmeans2`,
+      `scipy.interpolate.BarycentricInterpolator`,
+      `scipy.interpolate.barycentric_interpolate`,
+      `scipy.linalg.clarkson_woodruff_transform`,
+      `scipy.optimize.basinhopping`,
+      `scipy.optimize.differential_evolution`, `scipy.optimize.dual_annealing`,
+      `scipy.optimize.check_grad`, `scipy.optimize.quadratic_assignment`,
+      `scipy.sparse.random`, `scipy.sparse.random_array`, `scipy.sparse.rand`,
+      `scipy.sparse.linalg.svds`, `scipy.spatial.transform.Rotation.random`,
+      `scipy.spatial.distance.directed_hausdorff`,
+      `scipy.stats.goodness_of_fit`, `scipy.stats.BootstrapMethod`,
+      `scipy.stats.PermutationMethod`, `scipy.stats.bootstrap`,
+      `scipy.stats.permutation_test`, `scipy.stats.dunnett`, all
+      `scipy.stats.qmc` classes that consume random numbers, and
+      `scipy.stats.sobol_indices`.
+    - When passed by keyword, the ``rng`` argument will follow the SPEC 7
+      standard behavior: the argument will be normalized with
+      ``np.random.default_rng`` before being used.
+    - When passed by position or legacy keyword, the behavior of the argument
+      will remain unchanged (for now).
+
+  - Beginning in SciPy ``1.17.0``:
```

Comment:
I suggest to remove this, or reword briefly like _"It is planned that in 1.17.0 the legacy argument will start emitting warnings, and that in 1.19.0 the default behavior will change"_,  link to an issue where folks can see details and follow along. Plans for future releases often change, so this doesn't really belong in release notes.